docs/rules/object-key-newlines.md

```diff
@@ -0,0 +1,68 @@
+# Enforce placing object keys on separate lines (object-key-newlines)
+
+While formatting preferences are very personal, a number of style guides require that
```

Comment:
@nzakas What do you think about object-property-newlines for the rule identifier?

If I am thinking correctly about the if test `lastTokenOfPreviousValue.loc.end.line === firstTokenOfCurrentKey.loc.start.line` in https://github.com/eslint/eslint/pull/5933/files#diff-78f3e91fea2745f4c2bf92d1e39f7214R60 the following is incorrect (as you would desire if you have this stylistic guideline) even though the keys `a` and `d` are on different lines:

```
let obj = {
    a: {
        b: 1,
        c: 2
    }, d: 2
};
```